File: solutions/clojure/leap/1/src/leap.clj
(ns leap)

(defn leap-year?
  "Determine if a year is a leap year"
  [year]
  (if (= 0 (mod year 400))
    true
    (if (= 0 (mod year 100))
      false
      (= 0 (mod year 4))))
  )

File: solutions/cpp/leap/1/leap.cpp
#include "leap.h"

namespace leap {

  bool is_leap_year(int year) {
    if (year % 400 == 0) {
      return true;
    } else if (year % 100 == 0) {
      return false;
    } else {
      return year % 4 == 0;
    }
  }

}  // namespace leap
